### PR TITLE
Add @icon annotation definition

### PR DESCRIPTION
**Pull request by rdhananjaya**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/23696

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal [hunk 1]
```diff
@@ -68,3 +68,13 @@ public type StrandData record {|
 # Denotes new Strand execution semantics.
 public const annotation StrandData strand on source worker;
 
+
```
Comment (MaryamZi) [inline langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal:71]: Extra new line.

File: langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal [hunk 2]
```diff
@@ -68,3 +68,13 @@ public type StrandData record {|
...
+# Icon meta-data for types and functions.
+#
+# + path - relative filepath to module resource directory or url path to read the image.
+public type IconData record {|
+    string path;
+|};
+
+# icon annotation.
```
Comment (MaryamZi) [inline langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal:79]: Given that this is a public construct, shouldn't we have proper documentation?

File: langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal [hunk 3]
```diff
@@ -68,3 +68,13 @@ public type StrandData record {|
 # Denotes new Strand execution semantics.
 public const annotation StrandData strand on source worker;
 
+
+# Icon meta-data for types and functions.
+#
+# + path - relative filepath to module resource directory or url path to read the image.
+public type IconData record {|
```
Comment (hasithaa) [inline langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal:75]: Can we define IconData inline with the annotation?

File: langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal [hunk 4]
```diff
@@ -68,3 +68,5 @@ public type StrandData record {|
 # Denotes new Strand execution semantics.
 public const annotation StrandData strand on source worker;
 
+# This annotation specify icon meta-data about types and functions.
```
Comment (MaryamZi) [inline langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal:71]: For consistency should we say something like

```suggestion
# Denotes icon metadata related to types and functions.
```

File: tests/jballerina-unit-test/src/test/resources/test-src/annotations/icon_annot.bal [hunk 5]
```diff
@@ -0,0 +1,12 @@
+@icon { path: "/fooIconPath.icon" }
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/annotations/icon_annot.bal:1]: Missing licence headers.

Comment (rdhananjaya) [reply tests/jballerina-unit-test/src/test/resources/test-src/annotations/icon_annot.bal:17]: Added

File: langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal (hunk 4 above)
Comment (rdhananjaya) [reply langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal]: Fixed

**Comment by pubudu91** [comment]
Closing and reopening PR to retrigger the tests

**Pull request merged**